lessons/advanced/typespec.md

```diff
@@ -0,0 +1,197 @@
+---
+layout: page
+title: Specifications and types
+category: advanced
```

Comment:
Hmm, sorry I was late to the party when you changed this. I think it might be a good idea to change this to specific as it makes use of an external lib and fits with erlang. Dialyzer fits much closer with Mnesia, ETS / DETS, etc than gen_server, error handling etc.